### PR TITLE
README.md: Add cluster-machine-approver details.

### PR DESCRIPTION
**Pull request by russellb**
While working on integration of a new IPI platform type, I needed to
understand how automatic CSR approval worked, because it imposes some
expectations on each platform's cluster-api provider (or Machine
actuator).  This patch is adapted from some of my notes.

This expansion of the README aims to be a document that provides
background and pointers to important details for anyone looking
understand the CSR approval workflows.

### CODE REVIEW

**Comment by mrunalp** [comment]
:+1:

**Comment by mrunalp** [comment]
@enxebre @michaelgugino @vikaschoudhary16 ptal

File: README.md [hunk 1]
```diff
@@ -1,6 +1,129 @@
...
+First, there must be a `Machine` object with a `NodeRef` field set to the
+`Node` that sent this CSR.  The `NodeRef` is set by a `Node` controller under
+the [machine-api-operator](https://github.com/openshift/machine-api-operator).
+
+Once a `Node`-`Machine` pair has been identified, validation is done on all of
+the `Addresses` in the `Status` field of the `Machine`.  The CSR requests a
+certificate with the [SAN (Subject Alternate Names)
+extension](https://geekflare.com/san-ssl-certificate/).  The resulting
+certificate will be valid for every address or hostname listed on the `Node`
+resource To validate this request, the `cluster-machine-approver` ensures that
```
Comment (mrunalp) [inline README.md]: nit: To -> to

File: README.md [hunk 2]
```diff
@@ -1,6 +1,129 @@
...
+or [kube-controller-manager can be configured to approve
+them](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/#kube-controller-manager-configuration).
+Alternatively, some custom component could be built to approve CSRs through the
+API, which is what OpenShift has done.
+
+## OpenShift and CSRs
+
+OpenShift includes a custom component to approve CSRs: the
+`cluster-machine-approver`.  The `cluster-machine-approver` is used to
+automatically approve CSRs, but with more strict criteria that what was
```
Comment (mrunalp) [inline README.md]: that -> than

File: README.md (hunk 1 above)
Comment (russellb) [reply README.md]: thanks, fixed

File: README.md (hunk 2 above)
Comment (russellb) [reply README.md]: thanks, fixed

**Comment by russellb** [comment]
To see the rendered version of this PR: https://github.com/russellb/cluster-machine-approver/blob/docs/README.md

**Comment by bison** [comment]
I somehow missed this and went through the same process of discovery today... Looks great.
/approve

**Pull request merged**